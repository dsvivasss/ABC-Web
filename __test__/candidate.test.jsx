import React from 'react';
import { render } from '@testing-library/react';
import CandidateItem from '../src/app/candidate_detail/page';
import '@testing-library/jest-dom';

// Mock useRouter:
jest.mock("next/navigation", () => ({
    useRouter() {
      return {
        prefetch: () => null
      };
    }
  }));

describe('Candidate page', () => {
    it('Candidate Item page renders correctly', () => {
        const { container } = render(<CandidateItem />)
        expect(container).toMatchSnapshot()
    });
})