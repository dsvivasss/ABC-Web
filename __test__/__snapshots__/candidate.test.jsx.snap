// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candidate page Candidate Item page renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <nav
      class="bg-white"
      data-headlessui-state=""
    >
      <div
        class="mx-auto max-w-7xl px-1 sm:px-1 lg:px-1"
      >
        <div
          class="relative flex h-16 items-center justify-between"
        >
          <div
            class="absolute inset-y-0 left-0 flex items-center sm:hidden"
          >
            <button
              aria-expanded="false"
              class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
              data-headlessui-state=""
              id="headlessui-disclosure-button-:r0:"
              type="button"
            >
              <span
                class="absolute -inset-0.5"
              />
              <span
                class="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                class="block h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </div>
          <div
            class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
          >
            <div
              class="flex flex-shrink-0 items-center"
            />
          </div>
          <div
            class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
          >
            <button
              class="relative rounded-full bg-gray-800 p-1 text-gray-400 hover:text-white focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
              type="button"
            >
              <span
                class="absolute -inset-1.5"
              />
              <span
                class="sr-only"
              >
                Ver notifications
              </span>
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.857 17.082a23.848 23.848 0 005.454-1.31A8.967 8.967 0 0118 9.75v-.7V9A6 6 0 006 9v.75a8.967 8.967 0 01-2.312 6.022c1.733.64 3.56 1.085 5.455 1.31m5.714 0a24.255 24.255 0 01-5.714 0m5.714 0a3 3 0 11-5.714 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
            <div
              class="relative ml-3"
              data-headlessui-state=""
            >
              <div>
                <button
                  aria-expanded="false"
                  aria-haspopup="menu"
                  class="relative flex rounded-full bg-gray-800 text-sm focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
                  data-headlessui-state=""
                  id="headlessui-menu-button-:r1:"
                  type="button"
                >
                  <span
                    class="absolute -inset-1.5"
                  />
                  <span
                    class="sr-only"
                  >
                    Open user menu
                  </span>
                  <img
                    alt=""
                    class="h-8 w-8 rounded-full"
                    src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>
  <div
    class="flex"
  >
    <div
      class="bg-neutral-50 w-1/6"
    >
      <div
        class="min-h-screen flex flex-col flex-auto flex-shrink-0 antialiased bg-neutral-50 text-gray-800"
      >
        <div
          class="flex flex-col top-0 left-0 bg-neutral-50 h-full border-r"
        >
          <div
            class="overflow-y-auto overflow-x-hidden flex-grow"
          >
            <ul
              class="flex flex-col py-4 space-y-1"
            >
              <li
                class="px-5"
              >
                <div
                  class="flex flex-row items-center h-8"
                >
                  <div
                    class="text-sm font-light tracking-wide text-gray-500"
                  >
                    Menu
                  </div>
                </div>
              </li>
              <li>
                <a
                  class="flex items-center h-11 focus:outline-none hover:bg-gray-50 text-gray-600 hover:text-gray-800 border-l-4 border-transparent hover:border-indigo-500 pr-6"
                  href="http://localhost:3000/project_list"
                >
                  <span
                    class="inline-flex justify-center items-center ml-4"
                  >
                    <svg
                      class="w-5 h-5"
                      fill="none"
                      stroke="currentColor"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                  <span
                    class="ml-2 text-sm tracking-wide truncate"
                  >
                    Proyectos
                  </span>
                  <span
                    class="px-2 py-0.5 ml-auto text-xs font-medium tracking-wide text-green-500 bg-green-50 rounded-full"
                  >
                    3
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-5/6 p-4"
    >
      <div
        class="center2"
      >
        <div
          class="flex -space-x-2 overflow-hidden"
        >
          <img
            alt=""
            class="inline-block h-20 w-20 rounded-full ring-2 ring-white"
            src="https://images.unsplash.com/photo-1491528323818-fdd1faba62cc?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
          />
        </div>
      </div>
      <div
        class="center2"
      >
        <div
          class="grid3"
        >
          <div
            class="card"
          >
            <h1
              class="animate-fade-up text-2xl from-black bg-clip-text  font-bold leading-7 text-gray-900 sm:truncate sm:tracking-tight py-2"
            >
              Pedro Melenas
            </h1>
            <h1
              class="animate-fade-up text-xl from-black bg-clip-text  font-bold leading-7 text-gray-900 sm:truncate sm:tracking-tight py-2"
            >
              Descripción
            </h1>
            <p>
              Pedro Melenas es un distinguido ingeniero de software, destacado por sus contrubuciones a a la computación cuántica.
            </p>
            <h1
              class="animate-fade-up text-xl from-black bg-clip-text  font-bold leading-7 text-gray-900 sm:truncate sm:tracking-tight py-2"
            >
              Soft skills
            </h1>
            <span
              class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10"
            >
              Comunicación estratégica
            </span>
            <span
              class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10"
            >
              Trabajo en equipo
            </span>
            <span
              class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10"
            >
              Solución de problemas
            </span>
            <span
              class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10"
            >
              Creatividad
            </span>
            <span
              class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10"
            >
              Empatía
            </span>
            <h1
              class="animate-fade-up text-xl from-black bg-clip-text  font-bold leading-7 text-gray-900 sm:truncate sm:tracking-tight py-2"
            >
              Hard skills
            </h1>
            <span
              class="inline-flex items-center rounded-md bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >
              Python
            </span>
            <span
              class="inline-flex items-center rounded-md bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >
              Typescript
            </span>
            <span
              class="inline-flex items-center rounded-md bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >
              DevOps
            </span>
            <span
              class="inline-flex items-center rounded-md bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >
              Security
            </span>
            <span
              class="inline-flex items-center rounded-md bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >
              Software Architecture
            </span>
            <span
              class="inline-flex items-center rounded-md bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >
              Next Js
            </span>
            <h1
              class="animate-fade-up text-xl from-black bg-clip-text  font-bold leading-7 text-gray-900 sm:truncate sm:tracking-tight py-2"
            >
              Experiencia
            </h1>
            <div
              class="pb-2"
            >
              <dl
                class="divide-y divide-gray-100"
              >
                <div
                  class="px-4 py-2 sm:grid sm:grid-cols-1 sm:gap-4 sm:px-0"
                >
                  <dd
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-0"
                  >
                    Backend Developer
                  </dd>
                </div>
                <div
                  class="px-4 py-2 sm:grid sm:grid-cols-1 sm:gap-4 sm:px-0"
                >
                  <dd
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-0"
                  >
                    Senior Frontend Engineer
                  </dd>
                </div>
                <div
                  class="px-4 py-2 sm:grid sm:grid-cols-1 sm:gap-4 sm:px-0"
                >
                  <dd
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-0"
                  >
                    Software Architect
                  </dd>
                </div>
                <div
                  class="px-4 py-2 sm:grid sm:grid-cols-1 sm:gap-4 sm:px-0"
                >
                  <dd
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-0"
                  >
                    Backend Developer
                  </dd>
                </div>
              </dl>
            </div>
            <button
              class="flex justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
              type="submit"
            >
              <a
                href="/project_detail"
              >
                Seleccionar candidato
              </a>
            </button>
          </div>
          <div
            class="card"
          >
            <h1
              class="animate-fade-up text-2xl from-black bg-clip-text  font-bold leading-7 text-gray-900 sm:truncate sm:tracking-tight py-2"
            >
              Pruebas de selección
            </h1>
            <p>
              Este es un resumen de las pruebas que ha realizado el candidato.
            </p>
            <table
              class="table-auto divide-y divide-gray-300 py-3"
            >
              <thead
                class=""
              >
                <tr>
                  <th
                    class="text-lg from-black font-bold leading-2 text-gray-900 sm:truncate sm:tracking-tight py-1"
                  >
                    Prueba
                  </th>
                  <th
                    class="text-lg from-black font-bold leading-2 text-gray-900 sm:truncate sm:tracking-tight py-1"
                  >
                    Resultado
                  </th>
                </tr>
              </thead>
              <tbody
                class="divide-y divide-gray-300 py-2"
              >
                <tr>
                  <td
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-3 py-1"
                  >
                    Psicológica
                  </td>
                  <td
                    class="pl-5"
                  >
                    <span
                      class="inline-flex items-center rounded-md bg-blue-50 px-2 py-1 text-xs font-medium text-blue-700 ring-1 ring-inset ring-blue-700/10"
                    >
                      50%
                    </span>
                  </td>
                  <td>
                    <button
                      class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10 hover:bg-indigo-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    >
                      Modificar
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-3 py-1"
                  >
                    Idiomas
                  </td>
                  <td
                    class="pl-5"
                  >
                    <span
                      class="inline-flex items-center rounded-md bg-blue-50 px-2 py-1 text-xs font-medium text-blue-700 ring-1 ring-inset ring-blue-700/10"
                    >
                      90%
                    </span>
                  </td>
                  <td>
                    <button
                      class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10 hover:bg-indigo-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    >
                      Modificar
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-3 py-1"
                  >
                    Python
                  </td>
                  <td
                    class="pl-5"
                  >
                    <span
                      class="inline-flex items-center rounded-md bg-blue-50 px-2 py-1 text-xs font-medium text-blue-700 ring-1 ring-inset ring-blue-700/10"
                    >
                      80%
                    </span>
                  </td>
                  <td>
                    <button
                      class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10 hover:bg-indigo-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    >
                      Modificar
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="card"
          >
            <h1
              class="animate-fade-up text-2xl from-black bg-clip-text  font-bold leading-7 text-gray-900 sm:truncate sm:tracking-tight py-2"
            >
              Entrevistas
            </h1>
            <p
              class="pb-1"
            >
              Verifica las entrevistas realizadas al candidato.
            </p>
            <table
              class="table-auto divide-y divide-gray-300 py-3"
            >
              <thead
                class=""
              >
                <tr>
                  <th
                    class="text-lg from-black font-bold leading-2 text-gray-900 sm:truncate sm:tracking-tight py-1"
                  >
                    Fecha y hora
                  </th>
                  <th
                    class="text-lg from-black font-bold leading-2 text-gray-900 sm:truncate sm:tracking-tight py-1"
                  >
                    Resultado
                  </th>
                </tr>
              </thead>
              <tbody
                class="divide-y divide-gray-300 py-2"
              >
                <tr>
                  <td
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-3 py-1"
                  >
                    10 de Oct 2023
                  </td>
                  <td
                    class="pl-5"
                  >
                    <span
                      class="inline-flex items-center rounded-md bg-blue-50 px-2 py-1 text-xs font-medium text-blue-700 ring-1 ring-inset ring-blue-700/10"
                    >
                      50%
                    </span>
                  </td>
                  <td>
                    <button
                      class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10 hover:bg-indigo-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    >
                      Modificar
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-3 py-1"
                  >
                    20 de Oct 2023
                  </td>
                  <td
                    class="pl-5"
                  >
                    <span
                      class="inline-flex items-center rounded-md bg-blue-50 px-2 py-1 text-xs font-medium text-blue-700 ring-1 ring-inset ring-blue-700/10"
                    >
                      90%
                    </span>
                  </td>
                  <td>
                    <button
                      class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10 hover:bg-indigo-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    >
                      Modificar
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-3 py-1"
                  >
                    3 de Ago 2022
                  </td>
                  <td
                    class="pl-5"
                  >
                    <span
                      class="inline-flex items-center rounded-md bg-blue-50 px-2 py-1 text-xs font-medium text-blue-700 ring-1 ring-inset ring-blue-700/10"
                    >
                      80%
                    </span>
                  </td>
                  <td>
                    <button
                      class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10 hover:bg-indigo-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    >
                      Modificar
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-sm font-medium leading-6 text-gray-700 sm:px-3 py-1"
                  >
                    1 Feb de 2023
                  </td>
                  <td
                    class="pl-5"
                  >
                    <span
                      class="inline-flex items-center rounded-md bg-blue-50 px-2 py-1 text-xs font-medium text-blue-700 ring-1 ring-inset ring-blue-700/10"
                    >
                      20%
                    </span>
                  </td>
                  <td>
                    <button
                      class="inline-flex items-center rounded-md bg-indigo-50 px-2 py-1 text-xs font-medium text-indigo-700 ring-1 ring-inset ring-indigo-700/10 hover:bg-indigo-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    >
                      Modificar
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
